src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -358,14 +358,19 @@ private JObject GenerateDependencies(JObject projectJsonRoot, ITaskItem[] extern
                 {
                     NuGetVersion nuGetVersion = NuGetVersion.Parse(dependency.GetMetadata("Version"));
                     PackageItem packageItem = new PackageItem(name, nuGetVersion);
+                    bool? forceVersion = dependency.GetMetadata("ForceVersion")?.Equals("true", StringComparison.OrdinalIgnoreCase);
```

Comment:
What are you forcing the version to be and why?  ie, why would you not want to use the version provided by a package drop or version file and would instead prefer the assembly version?